src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -1937,6 +1937,104 @@ class spell_q13086_cannons_target : public SpellScriptLoader
+    NPC_RISEN_ALLIANCE_SOLDIERS           = 31205,
+    NPC_VICIOUS_GEIST                     = 31147,
+    NPC_RENIMATED_ABOMINATION             = 31692,
+
+    SPELL_ICY_GHOUL_CREDIT                = 59591, // Credit for Icy Ghoul
+    SPELL_VICIOUS_GEISTS_CREDIT           = 60042, // Credit for Vicious Geists
+    SPELL_RISEN_ALLIANCE_SOLDIERS_CREDIT  = 60040, // Credit for Risen Alliance Soldiers
+};
+
+class spell_q13264_thats_abominable : public SpellScriptLoader
+{
+    public:
+        spell_q13264_thats_abominable() : SpellScriptLoader("spell_q13264_thats_abominable") { }
+
+        class spell_q13264_thats_abominable_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_q13264_thats_abominable_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo(
+                {
+                    SPELL_ICY_GHOUL_CREDIT,
+                    SPELL_VICIOUS_GEISTS_CREDIT,
+                    SPELL_RISEN_ALLIANCE_SOLDIERS_CREDIT,
+                });
+            }
+
+            void HandleKnockBack(SpellEffIndex effIndex)
+            {
+                PreventHitDefaultEffect(effIndex);
+
+                if (Unit* unit = GetHitUnit())
+                {
+                    ObjectGuid objGuid = unit->GetGUID();
+
+                    // Except Player
+                    if (objGuid.IsPlayer())
+                        return;
+
+                    if (Unit* charmer = GetCaster()->GetCharmerOrOwner())
+                        if (Player* player = charmer->ToPlayer())
+                            if (player->GetQuestStatus(QUEST_THATS_ABOMINABLE) == QUEST_STATUS_INCOMPLETE)
+                                if (GiveCreditIfValid(player, &objGuid))
+                                    unit->KillSelf();
+                }
+            }
+
+            bool GiveCreditIfValid(Player* player, ObjectGuid* obj)
+            {
+                uint32 entry = obj->GetEntry();
+
+                switch(entry)
+                {
+                    case NPC_ICY_GHOUL:
+                        player->CastSpell(player, SPELL_ICY_GHOUL_CREDIT, true);
+                        return true;
+                        break;
+
+                    case NPC_VICIOUS_GEIST:
+                        player->CastSpell(player, SPELL_VICIOUS_GEISTS_CREDIT, true);
+                        return true;
+                        break;
+
+                    case NPC_RISEN_ALLIANCE_SOLDIERS:
+                        player->CastSpell(player, SPELL_RISEN_ALLIANCE_SOLDIERS_CREDIT, true);
+                        return true;
+                        break;
+                    default:
+                        return false;
+                        break;
+                }
+            }
+
+            void HandleScript(SpellEffIndex /*effIndex*/)
+            {
+                GetCaster()->ToCreature()->KillSelf();
```

Comment:
what if this spell is cast by a Player ?

Reply:
server crash ;)